File: front/src/ts/data/Person.ts
import Color from "./Color";
import Location from "./Location";
import Country from "./Country";

class Person {
    constructor(hairColor: Color, location: Location | null, passportID: string, nationality: Country | null, eyeColor: Color | null) {
        this.eyeColor = eyeColor;
        this.hairColor = hairColor;
        this.location = location;
        this.passportID = passportID;
        this.nationality = nationality;
    }

    eyeColor: Color | null
    hairColor: Color
    location: Location | null
    passportID: string
    nationality: Country | null

    public validate() {
        let valid = true

        if (this.passportID.length < 6) valid = false

        return valid
    }
}

export default Person

File: IS/back/src/main/java/com/example/back/data/Person.java
package com.example.back.data;

public class Person {
    private Color eyeColor; //Поле может быть null
    private Color hairColor; //Поле не может быть null
    private Location location; //Поле может быть null
    private String passportID; //Длина строки должна быть не меньше 6, Значение этого поля должно быть уникальным, Поле не может быть null
    private Country nationality; //Поле может быть null
}